### ISSUE TITLE
Unnecessary ref safety error reported for deconstruction

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#64448]
Compile:
```csharp
ref struct R
{
    public R(ref int i) { }
    public void Deconstruct(out int x, out int y) => throw null;
}

class Program
{
    static void Main()
    {
        int i = 0;
        var r = new R(ref i);

        int x, y;
        (x, y) = r;                  // error CS8352: 'r' ... may expose referenced variables
        r.Deconstruct(out x, out y); // ok
    }
}
```
Expected: No errors

Actual:
```
(15, 18): error CS8352: Cannot use variable 'r' in this context because it may expose
    referenced variables outside of their declaration scope
```

**Comment by cston** [comment]
See related https://github.com/dotnet/roslyn/issues/64000.

**Comment by cston** [comment]
A similar issue exists with 3.11:
```csharp
using System;

static class Program
{
    static void Main()
    {
        Span<int> s = stackalloc int[10];
        int x, y;
        (x, y) = s; // error CS8352: ...
        s.Deconstruct(out x, out y);
    }

    static void Deconstruct(this Span<int> s, out int x, out int y) => throw null;
}
```

**Comment by cston** [comment]
Closing as duplicate of #64000.
